### PR TITLE
chore: Cases optimisation

### PR DESCRIPTION
**Pull request by jfaracik**
The `/cases/` page is awful. Let's fix that.

∞ queries down to < 30 queries (excluding flags - flag optimisation will come in the flags colours story).

### CODE REVIEW

File: cases/fields.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        else:
+            return queryset.all()
+
+
+class HasOpenECJUQueriesRelatedField(PrimaryKeyRelatedField):
+    """
+    Custom field serializer required to determine if a case has open ECJU queries
+    """
+
+    def use_pk_only_optimization(self):
```
Comment (Mark-Spiers-kainos) [inline cases/fields.py:34]: I had to look this up to understand it, may be worthwhile to show to people?

Comment (jfaracik) [reply cases/fields.py:34]: Sure!

File: conf/middleware.py [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+
 from django.db import connection
 
+from conf.settings import env
+from test_helpers.colours import bold
+
+
+#
+#
+# Reset to master copy when done
```
Comment (Mark-Spiers-kainos) [inline conf/middleware.py:14]: remember to set back

File: cases/fields.py (hunk 1 above)
Comment (efahy) [reply cases/fields.py:34]: A comment wouldn't hurt either

**Pull request merged**